From 55f26caf5ee98152f2f14e85b21c4ba76fa8d41e Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Mon, 12 Feb 2018 08:44:57 +0700
Subject: [PATCH] Problem: source code is not highlighted in issues/comments

Solution: hook prims-highlighter to issue/rendered-text

It will load syntax files dynamically. Sometimes user
experience of that is less than optimal, but it is a good
enough first go, I believe.
---
 sit-web/webapp/src/issue/rendered-text.html | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/sit-web/webapp/src/issue/rendered-text.html b/sit-web/webapp/src/issue/rendered-text.html
index a84ec57..5dada94 100644
--- a/sit-web/webapp/src/issue/rendered-text.html
+++ b/sit-web/webapp/src/issue/rendered-text.html
@@ -1,11 +1,18 @@
 <link rel="import" href="/bower_components/polymer/polymer-element.html">
 <link rel="import" href="/bower_components/marked-element/marked-element.html">
+<link rel="import" href="/bower_components/prism-element/prism-highlighter.html">
+<link rel="import" href="/bower_components/iron-ajax/iron-ajax.html">
 
 <dom-module id="rendered-text">
    <template>
 
        <link rel="stylesheet" href="/src/issue/rendered-text.css">
+       <link rel="stylesheet" href="/bower_components/prism/themes/prism.css">
 
+       <iron-ajax auto url="/bower_components/prism/components/prism-[[lang]].js"
+                  handle-as="text" on-response="loadSyntax"></iron-ajax>
+
+       <prism-highlighter></prism-highlighter>
        <marked-element markdown="{{ text }}">
            <div slot="markdown-html"></div>
        </marked-element>
@@ -18,9 +25,28 @@
                 return {
                     text: {
                         type: String
+                    },
+                    lang: {
+                        type: String
                     }
                 }
             }
+            constructor() {
+                super();
+                this.lang = "markdown";
+            }
+            loadSyntax(evt, req) {
+                eval(req.response);
+                // New String is allocated so that an
+                // update is triggered
+                this.set("text", new String(this.text));
+            }
+            ready() {
+                this.addEventListener('syntax-highlight', function(e) {
+                   this.set("lang", e.detail.lang);
+                });
+                super.ready();
+            }
         }
         customElements.define(RenderedText.is, RenderedText);
     </script>
-- 
2.15.0

